h/services/search_index/_queue.py

```diff
@@ -23,30 +25,31 @@ def __init__(self, db, es, batch_indexer):
         self._es = es
         self._batch_indexer = batch_indexer
 
-    def add(self, annotation_id, tag, schedule_in=None):
-        """Queue an annotation to be synced to Elasticsearch."""
-        self.add_all([annotation_id], tag, schedule_in)
-
-    def add_all(self, annotation_ids, tag, schedule_in=None):
-        """Queue a list of annotations to be synced to Elasticsearch."""
-
-        scheduled_at = (datetime.utcnow() + schedule_in) if schedule_in else None
+    def add_annotations_between_times(self, start_time, end_time, tag):
+        """
+        Add all annotations between two times to the search index.
 
-        # Jobs with a lower number for their priority get processed before jobs
-        # with a higher number. Make large batches of jobs added all at once
-        # get processed *after* small batches added a few at a time, so that
-        # large batches don't hold up small ones for a long time.
-        priority = len(annotation_ids)
+        All annotations whose updated time is between the given start_time and
+        end_time (inclusive) will be queued for background indexing into
+        Elasticsearch.
 
-        self._db.add_all(
-            Job(
-                tag=tag,
-                name="sync_annotation",
-                scheduled_at=scheduled_at,
-                priority=priority,
-                kwargs={"annotation_id": annotation_id},
+        :type start_time: datetime.datetime
+        :type end_time: datetime.datetime
+        """
+        self._db.execute(
+            Job.__table__.insert().from_select(
+                [Job.name, Job.priority, Job.tag, Job.kwargs],
+                select(
+                    [
+                        text("'sync_annotation'"),
+                        text("1000"),
+                        text(repr(tag)),
+                        func.jsonb_build_object("annotation_id", Annotation.id),
+                    ]
+                )
+                .where(Annotation.updated >= start_time)
+                .where(Annotation.updated <= end_time),
```

Comment:
I think this is probably more or less as efficient as we're gonna get, in terms of adding annotations to the jobs table. So hopefully this is fast enough to reindex large timeframes

Reply:
If it's not quick enough, we can always spin it off into an async task. Nicer if we don't have to bother

Reply:
If it's not quick enough, we can always spin it off into an async task. Nicer if we don't have to bother